### PR TITLE
feat: Main menu

### PR DESCRIPTION
**Pull request by ramabit**
# Description

https://www.notion.so/thebadge/Create-sidebar-component-no-logic-5d84d45fdc9c486f8684ae3e98ae4099 

## What's new?

- Base implementation of new main menu

# Screenshots
Screenshot 2023-01-23 at 19 27 25

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -61,12 +77,15 @@ export default function App({
...
-                  <TransactionNotificationProvider>
-                    <CookiesWarningProvider>
-                      {getLayout(<Component {...pageProps} />)}
-                      <Footer />
-                    </CookiesWarningProvider>
-                  </TransactionNotificationProvider>
+                  <SectionReferencesProvider>
+                    <TransactionNotificationProvider>
+                      <CookiesWarningProvider>
+                        <BackgroundGradient gradient={gradients.gradientBackground} />
```
Comment (FedeMadoery) [inline pages/_app.tsx:83]: What's the advantage of having it here? Also the footer should be inside, that the footer is not related to this PR, just saying

File: src/components/navigation/MainMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+      return colors.greenLogo
+    case 'gray':
+      return '#BDBDBD'
+    case 'small':
+    default:
+      return 'transparent'
+  }
+}
+
+const getMenuItemHoverBackgroundColor = (type: MenuItemType): string => {
```
Comment (FedeMadoery) [inline src/components/navigation/MainMenu.tsx:106]: What about moving this function to a `.utils`? Same for the routes below, to keep the component clean (this comp may end on the UI-LIb but well, just saying)

File: src/components/navigation/MobileMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const Wrapper = styled('div')(({ theme }) => ({
+  // background-color: ${({ theme }) => theme.modal.overlayColor};
+  bottom: 0,
+  left: 0,
+  position: 'fixed',
+  // top: ${({ theme }) => theme.header.height},
+  width: '100vw',
+  zIndex: 5,
+
+  // @media (min-width: ${({ theme }) => theme.breakPoints.tabletLandscapeStart}) {
```
Comment (FedeMadoery) [inline src/components/navigation/MobileMenu.tsx:20]: If we can remove the commented code, would be nice 🧹

File: src/components/navigation/MainMenu.tsx (hunk 2 above)
Comment (ramabit) [reply src/components/navigation/MainMenu.tsx:106]: you mean a MainMenu.utils right?

Comment (FedeMadoery) [reply src/components/navigation/MainMenu.tsx:106]: Yeah, that's it

File: src/components/navigation/MainMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+} from '@/src/components/navigation/Menu.types'
+import { useSectionReferences } from '@/src/providers/referencesProvider'
+import { useColorMode } from '@/src/providers/themeProvider'
+import { useWeb3Connection } from '@/src/providers/web3ConnectionProvider'
+import { ThemeType } from '@/src/theme/types'
+
+const MenuContainer = styled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  background: '#F4F4F4',
```
Comment (FedeMadoery) [inline src/components/navigation/MainMenu.tsx:27]: You may want to use the `theme. pallete.background.default` and then we can put our wanted color on it. But with this, you will be sure that the background is according to the theme mode.

File: src/components/navigation/MainMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+    : null),
+}))
+
+const SubMenuTitleItem = styled('div')(({ theme }) => ({
+  cursor: 'pointer',
+  fontWeight: 900,
+  fontSize: '12px',
+  lineHeight: '15px',
+  textTransform: 'uppercase',
+  color: '#000000',
```
Comment (FedeMadoery) [inline src/components/navigation/MainMenu.tsx:164]: Same here, you may want to use `theme.pallete.text.primary`, so the theme mode will define the right color for it.

File: src/components/navigation/MainMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+  },
+}))
+
+const SubMenuItem = styled('div')(({ theme }) => ({
+  cursor: 'pointer',
+  fontWeight: 600,
+  fontSize: '11px',
+  lineHeight: '14px',
+  textTransform: 'uppercase',
+  color: '#4f4f4f',
```
Comment (FedeMadoery) [inline src/components/navigation/MainMenu.tsx:180]: Same as TitleItem

**Comment by FedeMadoery** [comment]
I list some functional aspects that I found useful to have:

- The menu should close itself if you click outside of it
- The menu should be outside of the "_container_"
- The menu should grow in height according to the screen size

We may validate this with Lu, but I think she will agree.

File: src/components/navigation/MainMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+  '&:hover': {
+    opacity: 0.7,
+    borderBottom: '1px solid rgba(0, 0, 0, 0.7)',
+  },
+}))
+
+const SubMenuItem = styled('div')(({ theme }) => ({
+  cursor: 'pointer',
+  fontWeight: 600,
```
Comment (FedeMadoery) [inline src/components/navigation/MainMenu.tsx:176]: I feel like this should be 500, not sure why but all the weights are not one-to-one with Figma 🤔 ... Anyways we can share it with Lu and see what she saids

**Review by FedeMadoery — APPROVED** [review]
